tests/settings/ConfigurationGatheringTest.py

```diff
@@ -240,6 +240,16 @@ def test_targets(self):
 
         self.assertEqual(targets, ['cli', 'test1', 'test2'])
 
+    def test_exclude_sections(self):
+        current_dir = os.path.abspath(os.path.dirname(__file__))
+        test_dir = os.path.join(current_dir, 'section_manager_test_files')
+        with change_directory(test_dir):
+            sections, local_bears, global_bears, targets = gather_configuration(
+                lambda *args: True,
+                self.log_printer,
+                arg_list=['/!all.python', '-c', 'inherit_coafile'])
```

Comment:
Why the `/` ?

Reply:
> But, we already use positional arguments for target section names.

So then these are errors:

coala --exclude-sections foo foo
coala --exclude-sections foo bar
i.e. there is no valid value for TARGETS when --exclude-sections is used, and the inverse is also true.

IMO it would be more sensible for TARGETS to be prefixed with ! to indicate they should be disabled. i.e.

coala '!bar'
Then it is easier to check and document that TARGETS do not have conflicting sets of values like coala bar !bar



jayvdb had commented this on the issue.So I tried to implement the same

Reply:
Didn't see where he says "`/!`".

Is "`/`" required in the test only?

Reply:
Oh sorry my bad ... It's for escape character as bash shows the error 
`event not found`
when only exclamation point is used.I changed it to backslash in the current commit.Sorry for the error :sweat_smile:

Reply:
@blazeu  I'm not able to understand why the tests are failing.The AppVeyor console is not showing any specific error message